packages/extension/src/view/devtools/stateProviders/settings/settingsProvider.tsx

```diff
@@ -214,20 +210,64 @@ const Provider = ({ children }: PropsWithChildren) => {
     }
   }, [settingsChanged]);
 
+  useEffect(() => {
+    if (!isEqual(isUsingCDP, isUsingCDPForSettingsPageDisplay)) {
+      setSettingsChanged(true);
+    } else {
+      setSettingsChanged(false);
+      chrome.storage.session.remove(['pendingReload', 'isUsingCDP']);
+    }
+  }, [isUsingCDP, isUsingCDPForSettingsPageDisplay]);
+
+  const onCommittedNavigationListener = useCallback(
+    ({
+      frameId,
+      frameType,
+      url,
+      tabId,
+    }: chrome.webNavigation.WebNavigationTransitionCallbackDetails) => {
+      if (frameType !== 'outermost_frame' && frameId !== 0) {
+        return;
+      }
+
+      if (url.startsWith('chrome') || url.startsWith('devtools')) {
+        return;
+      }
+
+      if (!url) {
+        return;
+      }
+
+      if (tabId !== chrome.devtools.inspectedWindow.tabId) {
+        return;
+      }
```

Comment:
```suggestion
    const isNotTopLevelFrame = frameType !== 'outermost_frame' && frameId !== 0;
    const isInternalChromeURL = url?.startsWith('chrome') || url?.startsWith('devtools');
    const isNotCurrentTab = tabId !== chrome.devtools.inspectedWindow.tabId;

    if (isNotTopLevelFrame || isInternalChromeURL || !url || isNotCurrentTab) {
      return;
    }
```

to make it more readable.